pep-3156.txt

```diff
@@ -943,6 +970,52 @@ traceback.  In some cases they are caught and re-raised.  (Examples of
 this category include ``KeyboardInterrupt`` and ``SystemExit``; it is
 usually unwise to treat these the same as most other exceptions.)
 
+The event loop passes caught but not handled exceptions into own's
+*exception handler*.  The handler is a callback which accepts
+*context* dict as a parameter::
+
+   def exception_handler(context):
+       ...
+
+TBD: describe context's keys.
+
+The loop has the following methods related to exception handling:
+
+- ``get_exception_handler()`` returns the current handler registered
+  in the loop.
+
+- ``set_exception_handler(handler)`` assings new handler.
+
+- ``default_exception_handler(context)`` is *default* handler for the
+  loop implementation.
+
+- ``call_exception_handler(context)`` passes *context* into the
+  registered exception handler. The call allows to handle uncaught
+  exceptions uniformly by third-party libraries.
+
+  The loop uses ``default_exception_handler()`` if the default was not
+  overridden by explicit ``set_exception_handler()`` call.
+
+Debug Mode
+----------
+
+By default the loop operates in *release* mode for sake of performance.
+But application may enable *debug* mode.
+
+In this mode many additional checks are enabled, for example:
+
+- source tracebacks are available for unrolled exceptions in futures/tasks.
+
+- loop checks callback execution time for blaming blocking code.
```

Comment:
Ah, cool, I'm glad the default asyncio event loops have this feature now. (I had been meaning to contribute it since it's the one thing that I'd miss if Tornado switched to asyncio's event loop by default). This check is parameterized (how long is "too long" for blocking?); should we codify a way to pass extra parameters here? (e.g. declaring `set_debug()` to take `**kwargs` in addition to a bool) Of course, not all event loop implementations will support all debug-mode features so we'd need to either require that unknown keyword arguments be ignored or recommend that applications only use debug-mode keyword arguments when they know what event loop implementation is in use.